contrib/mesos/pkg/executor/executor.go

```diff
@@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
 	// (a) we ignore future launches for it
 	// (b) we have a record of it so that we can kill it if needed
 	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
+	var launchTimer *time.Timer
+	if k.launchGracePeriod > 0 {
+		launchTimer = time.NewTimer(k.launchGracePeriod)
+	} else {
+		launchTimer = &time.Timer{}
```

Comment:
because launchGracePeriod == 0 and in that case I want launchTimer.C to be
nil

On Tue, Dec 8, 2015 at 7:03 PM, Karl Isenberg <notifications@github.com>
wrote:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/17917#discussion_r47034353>
> :
>
> > @@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
> >  	// (a) we ignore future launches for it
> >  	// (b) we have a record of it so that we can kill it if needed
> >  	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
> > +	var launchTimer *time.Timer
> > +	if k.launchGracePeriod > 0 {
> > +		launchTimer = time.NewTimer(k.launchGracePeriod)
> > +	} else {
> > +		launchTimer = &time.Timer{}
>
> why the empty timer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17917/files#r47034353>.
>